src/components/Banners/SiteBanner.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from "react";
+import { useQuery } from "@apollo/react-hooks";
+import styled from "@emotion/styled";
+import { uiColors } from "@leafygreen-ui/palette";
+import IconButton from "@leafygreen-ui/icon-button";
+import Icon from "@leafygreen-ui/icon";
+import Cookies from "js-cookie";
+import get from "lodash.get";
+import { GET_SITE_BANNER } from "gql/queries";
+import { SiteBannerQuery } from "gql/generated/types";
+
+const { yellow, blue, green, red } = uiColors;
+export const SiteBanner = () => {
+  const { data, loading } = useQuery<SiteBannerQuery>(GET_SITE_BANNER);
+  const { text, theme } = get(data, "siteBanner", {});
+  const [showBanner, setShowBanner] = useState(false);
+  useEffect(() => {
+    if (text !== "" && Cookies.get(text) === undefined) {
+      setShowBanner(true);
+    }
+  }, [text]);
+  if (loading) return null;
+
+  const hideBanner = () => {
+    // If a user sees a banner and closes it lets set a cookie with the banners text as the key.
+    // The cookie will be auto deleted after a week. This ensures if a new banner with different text is returned we dont accidently hide it
+    setShowBanner(false);
+    Cookies.set(text, "viewed", { expires: 7 });
```

Comment:
I can think of some edge cases that could result from deleting the cookie and using the text as an identifier for the banner if the text is not unique that you're probably aware of.
A user can miss a message if banner A, B, and then A are displayed in the same week but the user doesn't log in to see banner B, the user will miss seeing banner A the second time around. 
A user will see the same message if the same banner is being displayed after 7 days (or the user resets their browser which we can't control)
I think the only way to avoid the first edge case is to use a unique id for the banner. The second edge case can be avoided by adding a "lastBannerViewed" flag that has the id of the last banner the user viewed.

Reply:
I don't agree with that logic. 
The way it's currently implemented, If a user views banner A and dismisses it. We will not display another banner with the SAME text for 1 Week. 
If another Banner B is displayed the User will still see it, but if they dismiss it it wont show up for another week (If the banner is still there). If we decide to reintroduce a banner with the same text as Banner A and it is within 1 week of the First time banner A was dismissed we wont show it . (Chances are its the same issue that was present earlier in the week which the user already acknowledged.) 
I dont think we should necessarily worry about a user deleting cookies since if a user does that there will be other issues besides them having to see a banner one more time.
If the user deletes cookies they will be un authenticated from the site since we also store auth tokens in cookies. 
![image](https://user-images.githubusercontent.com/4605522/87066220-34bf2a80-c1e0-11ea-9546-c811e4dbb067.png)

The banner text serves as the Unique id in this case since each banner will have its own text.
IMO It seems overkill to store whether or not a user viewed a banner on the db, Especially since the banners are only ever there for a short period of time.

Reply:
thanks for the explanation - I was wrong about the first edge case I described. I was under the impression that we only ever use one cookie for this feature but I realize we create a cookie per unique banner text. I also think it makes sense to not accommodate for the second edge case if it's unlikely the same banner will be active after 7 days.